api_end_end_testing/src/test/java/tds/user/api/tests/UserApiTests.java

```diff
@@ -0,0 +1,359 @@
+        .then()
+            .statusCode(204);
+
+        // Execute a DELETE to delete the user
+        deleteExistingUser(randomUserEmail);
+
+        // Execute a GET by email to validate user is deleted
+        given()
+            .contentType(ContentType.JSON)
+            .accept(ContentType.JSON)
+            .header(authHeader)
+            .when()
+            .get(uriLocation + "/" + randomUserEmail + "/details")
+            .then()
+            .statusCode(404);
+    }
+
+    /*
+     * Test of Create User with invalid email, HTTP POST of /rest/external/user, 400 bad request
+     */
+    @Test
+    public void shouldNotCreateUserWithInvalidEmail() {
+        List<RoleAssociation> roleAssociations = new ArrayList<RoleAssociation>();
+        roleAssociations.add(new RoleAssociation("Administrator", "CLIENT", "19037"));
+
+        UserInfo userInfo = new UserInfo("bademail", "Judy", "Bloom", "808-443-1199", roleAssociations);
+
+        given()
+            .contentType(ContentType.JSON)
+            .header(authHeader)
+            .body(userInfo)
+            .when()
+            .post(uriLocation)
+            .then()
+            .statusCode(400)
+            .body("messages.email[0]", equalTo("User Email Address must be a valid email address"));
+    }
+
+    /*
+     * Test of Create User with invalid first name, HTTP POST of /rest/external/user, 400 bad request
+     */
+    @Test
+    public void shouldNotCreateUserWithInvalidFirstName() {
+        String randomUserEmail = createRandomUserEmail();
+
+        List<RoleAssociation> roleAssociations = new ArrayList<RoleAssociation>();
+        roleAssociations.add(new RoleAssociation("Administrator", "CLIENT", "55342"));
+
+        UserInfo userInfo = new UserInfo(randomUserEmail, "", "Dodge", "714-228-4848", roleAssociations);
+
+        given()
+            .contentType(ContentType.JSON)
+            .header(authHeader)
+            .body(userInfo)
+            .when()
+            .post(uriLocation)
+            .then()
+            .statusCode(400)
+            .body("messages.firstName[0]", equalTo("User First Name is required"));
+    }
+
+    /*
+     * Test of Create User with invalid last name, HTTP POST of /rest/external/user, 400 bad request
+     */
+    @Test
+    public void shouldNotCreateUserWithInvalidLastName() {
+        String randomUserEmail = createRandomUserEmail();
+
+        List<RoleAssociation> roleAssociations = new ArrayList<RoleAssociation>();
+        roleAssociations.add(new RoleAssociation("Administrator", "CLIENT", "55342"));
+
+        UserInfo userInfo = new UserInfo(randomUserEmail, "Wiliam", "", "714-228-4848", roleAssociations);
+
+         given()
+            .contentType(ContentType.JSON)
+            .header(authHeader)
+            .body(userInfo)
+        .when()
+            .post(uriLocation)
+        .then()
+            .statusCode(400)
+         .body("messages.lastName[0]", equalTo("User Last Name is required"));
```

Comment:
Should be indented at the same level as `statusCode()`. Along with`stausCode`, `body` is part of the response that's being evaluated.